src/Progress.UnitTests/ProgessController/VsOutputWindowPaneNotifierTests.cs

```diff
@@ -0,0 +1,155 @@
+
+            // 4th Argument invalid
+            Exceptions.Expect<ArgumentNullException>(() => new VsOutputWindowPaneNotifier(this.serviceProvider, this.outputWindowPane, false, null, false));
+            Exceptions.Expect<ArgumentNullException>(() => new VsOutputWindowPaneNotifier(this.serviceProvider, this.outputWindowPane, false, string.Empty, false));
+            Exceptions.Expect<ArgumentNullException>(() => new VsOutputWindowPaneNotifier(this.serviceProvider, this.outputWindowPane, false, " \t", false));
+
+            // Valid
+            new VsOutputWindowPaneNotifier(this.serviceProvider, this.outputWindowPane, true, "{0}", false);
+            new VsOutputWindowPaneNotifier(this.serviceProvider, this.outputWindowPane, false, "{0}", true);
+        }
+
+        [TestMethod]
+        [Description("Verifies notifying of an exception message using an output window")]
+        public void VsOutputWindowPaneNotifier_MessageOnly()
+        {
+            // Setup
+            bool logWholeMessage = true;
+            StubVsUIShell.StubWindowFrame frame = this.Setup(false, logWholeMessage);
+
+            // Execute
+            ((IProgressErrorNotifier)this.testSubject).Notify(this.generatedException);
+
+            // Verify
+            frame.AssertNotShown();
+            this.outputWindowPane.AssertNotActivated();
+            this.outputWindowPane.AssertWrittenToOutputWindow();
+        }
+
+        [TestMethod]
+        [Description("Verifies notifying of a full exception using an output window")]
+        public void VsOutputWindowPaneNotifier_FullException()
+        {
+            // Setup
+            bool logWholeMessage = true;
+            StubVsUIShell.StubWindowFrame frame = this.Setup(false, logWholeMessage);
+
+            // Execute
+            ((IProgressErrorNotifier)this.testSubject).Notify(this.generatedException);
+
+            // Verify
+            frame.AssertNotShown();
+            this.outputWindowPane.AssertNotActivated();
+            this.outputWindowPane.AssertWrittenToOutputWindow();
+        }
+
+        [TestMethod]
+        [Description("Verifies notifying of a full exception using an output window and ensure that the output is visible")]
+        public void VsOutputWindowPaneNotifier_EnsureOutputVisible()
+        {
+            // Setup
+            bool logWholeMessage = true;
+            StubVsUIShell.StubWindowFrame frame = this.Setup(true, logWholeMessage);
+
+            // Execute
+            ((IProgressErrorNotifier)this.testSubject).Notify(this.generatedException);
+
+            // Verify
+            frame.AssertShown();
+            this.outputWindowPane.AssertActivated();
+            this.outputWindowPane.AssertWrittenToOutputWindow();
+        }
+        #endregion
+
+        #region Helpers
+        
+        private StubVsUIShell.StubWindowFrame Setup(bool ensureOutputVisible, bool logWholeMessage)
```

Comment:
Was a copy-paste-update from the existing VsGeneralOutputWindow tests.
Will rework to make clearer.